corehq/apps/reports/filters/users.py

```diff
@@ -377,9 +390,10 @@ def user_es_query(cls, domain, mobile_user_and_group_slugs, request_user):
             )
 
         if HQUserType.ACTIVE in user_types or HQUserType.DEACTIVATED in user_types:
-            # return all users with selected user_types
-            user_type_filters.append(user_es.mobile_users())
-            return q.OR(*user_type_filters)
+            if has_user_ids:
```

Comment:
Would it possible to add tests for this change?

Reply:
Done.